homeassistant/components/signal_messenger_jsonrpc/notify.py

```diff
@@ -0,0 +1,199 @@
+"""Signal Messenger (JSONRPC) for notify component."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from pysignalclijsonrpc.api import SignalCliJSONRPCApi, SignalCliJSONRPCError
+from requests import Session
+import voluptuous as vol
+
+from homeassistant.components.notify import (
+    ATTR_DATA,
+    PLATFORM_SCHEMA,
+    BaseNotificationService,
+)
+from homeassistant.core import HomeAssistant
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_ACCOUNT = "account"
+CONF_RECICPIENTS = "recipients"
+CONF_SIGNAL_CLI_REST_API_URL = "url"
+CONF_SIGNAL_CLI_REST_API_USERNAME = "username"
+CONF_SIGNAL_CLI_REST_API_PASSWORD = "password"
+CONF_MAX_ALLOWED_DOWNLOAD_SIZE_BYTES = 52428800
+ATTR_FILENAMES = "attachments"
+ATTR_URLS = "urls"
+ATTR_VERIFY_SSL = "verify_ssl"
+
+DATA_FILENAMES_SCHEMA = vol.Schema({vol.Required(ATTR_FILENAMES): [cv.string]})
+
+DATA_URLS_SCHEMA = vol.Schema(
+    {
+        vol.Required(ATTR_URLS): [cv.url],
+        vol.Optional(ATTR_VERIFY_SSL, default=True): cv.boolean,
+    }
+)
+
+DATA_SCHEMA = vol.Any(
+    None,
+    DATA_FILENAMES_SCHEMA,
+    DATA_URLS_SCHEMA,
+)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_ACCOUNT): cv.string,
+        vol.Required(CONF_SIGNAL_CLI_REST_API_URL): cv.string,
+        vol.Optional(CONF_SIGNAL_CLI_REST_API_USERNAME, default=""): cv.string,
+        vol.Optional(CONF_SIGNAL_CLI_REST_API_PASSWORD, default=""): cv.string,
+        vol.Required(CONF_RECICPIENTS): vol.All(cv.ensure_list, [cv.string]),
+    }
+)
```

Comment:
For integrations that connect to devices or services, we no longer accept new YAML configurations or changes.

This integration needs to be refactored first to use a configuration flow and config entries first.

More information about this can be found in Architecture Decision Record:

- ADR-0010: <https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md#decision>

See our developer documentation on how to get started creating a configuration flow for this integration:

<https://developers.home-assistant.io/docs/config_entries_config_flow_handler>

As these changes often involve a bit of work and some significant shift in the current code, we will close this PR for now.

We (and our community!) would really appreciate it if you could start on the refactoring of this integration in a new PR.

Thanks already! :+1: